src/node/config.rs

```diff
@@ -0,0 +1,22 @@
+use serde::Deserialize;
+use std::net::SocketAddr;
+use std::str::FromStr;
+
+pub const DEFAULT_JSON_RPC_ADDR: &str = "127.0.0.1:3030";
+pub const DEFAULT_JSON_RPC_PROTOCOL: &str = "http";
```

Comment:
We should make these configurable either through the IPC agent config file, or through a CLI flag when we start the IPC agent deamon. I guess it makes more sense to adress it right away in a follow-up PR. Can you add an issue or prepare a PR to tackle this? Thanks!

Reply:
In the configuration, we have a filed to handle the user input. I was thinking the default should be when user has not input anything. If this is from config, then the question is what happens if user does not pass in those default addr or protocol? Then we are asking them to define both the default and the expected url?

Reply:
In the design doc we considered that the daemon always accepts the path of a config file (or use the default config location) as input: https://docs.google.com/document/d/14lkRRv6MQYnuEfp2GoGngdD8Q5YgfE38D8HTZWKgKf4/edit#heading=h.i71qsbf2iwn5

With this in mind I don't think the default address will ever be not set, as a config file should be picked up, even if it is the default one. In any case, I guess we can address this in the coming PR with the daemon command and discuss how to handle it there. WDYT?

Reply:
Yeah, I can use that config file. But in the file you sent, I dont see a config parameter for the addr of the current node? Or it's not added yet?

Reply:
Yes, that's a good point, @hmoniz. Let's call it `server` or if we want to be verbose (and just in case we end up with more future servers) we can use something like `AGENT_RPC_SERVER`

Reply:
Here is what I'm proposing:
Update existing `config.rs` into a mod, then the `Config` struct will stay in the `mod.rs`. The overall folder struct is something like:
```
config/
-- mod.rs
-- subnet.rs  // contains subnet related configs
-- server.rs  // contains server related configs
-- utils.rs  // contains all of the `deserialize_*` functions that can be shared.
```
This way maybe we can add more configs without problem. The other way which I was thinking of is actually putting the configs into each module, i.e. `server.rs` is placed in the module together with rpc node, so that all the code related to rpc node is contained in one folder. Both should be quick changes, I guess it's just a preference, maybe we try putting the configs in one folder first? 

@hmoniz @adlrocha any thoughts?

Reply:
I really like your first suggestion as it keeps all the config (1) self-contained and (2) neatly modularized.

Reply:
I like that approach of having a single config crate with topical files. I'll review the linked PR also. 🙏